src/identity-providers/add/DescriptorSettings.tsx

```diff
@@ -0,0 +1,428 @@
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { Controller, useFormContext, useWatch } from "react-hook-form";
+import {
+  ExpandableSection,
+  FormGroup,
+  Select,
+  SelectOption,
+  SelectVariant,
+  TextInput,
+  ValidatedOptions,
+} from "@patternfly/react-core";
+import { HelpItem } from "../../components/help-enabler/HelpItem";
+import { SwitchField } from "../component/SwitchField";
+import { TextField } from "../component/TextField";
+
+import "./discovery-settings.css";
+
+type DescriptorSettingsProps = {
+  readOnly: boolean;
+};
+
+const Fields = ({ readOnly }: DescriptorSettingsProps) => {
+  const { t } = useTranslation("identity-providers");
+  const { t: th } = useTranslation("identity-providers-help");
+
+  const { register, control, errors } = useFormContext();
```

Comment:
A generic type should be specified here to make the returned value type-safe.
```suggestion
  const { register, control, errors } = useFormContext<MyFormFields>();
```

Reply:
this did not work as expected, leaving as is for this iteration